ed25519/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+extern crate ewasm_api;
+extern crate ed25519_dalek;
+extern crate sha2;
+
+pub mod ed25519_verify;
+
+#[no_mangle]
+pub extern "C" fn main() {
+    let mut length = ewasm_api::calldata_size();
+    let mut zero_len: usize = 0;
+
+    if length < 128 {
+      zero_len = 128 - length;
+    } else {
+      length = 128;
+    }
+
+    ewasm_api::consume_gas(2000);
+
+    let mut input_arr: [u8; 128] = [0; 128];
+    let mut input = ewasm_api::unsafe_calldata_copy(0, length);
+    input.append(&mut vec![0; zero_len]);
+    input_arr.copy_from_slice(&input[..128]);
```

Comment:
It feels weird to have to copy because I am unable to coerce the slice into a fixed array type (my C brain at work).  Is there a better way of doing this?

Reply:
What are you trying to do here?

Reply:
I was under the impression that input should be zero-padded if it was too short.  Either the EIP changed or I was mistaken in this assumption.